### PR TITLE
tests: fix classic-prepare-image test

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Ruuning with su the command "snap prepare image" is making followins
tests fail when them try to access to
/run/user/0/snapd-session-agent.socket

This happens when the su -c command is used without the test user

To reproduce the isse run:
> spread-debug -order -debug -single-worker
google:ubuntu-20.04-64:tests/main/classic-prepare-image
google:ubuntu-20.04-64:tests/main/snap-user-service-start-on-install

Like this example
>2021-06-18 14:36:25 Error executing
google:ubuntu-20.04-64:tests/main/snap-user-service-start-on-install
(jun181711-404796) :
>echo 'Install the a snap with user services while a user session is
active'
>Install the a snap with user services while a user session is active
/home/gopath/src/github.com/snapcore/snapd/tests/lib/tools/snaps-state
>install-local test-snapd-user-service
error: cannot perform the following tasks:
>Make snap "test-snapd-user-service" (unset) available to the system
(Post http://0/v1/service-control: dial unix
/run/user/0/snapd-session-agent.socket: connect: connection refused)

### CODE REVIEW

File: tests/main/classic-prepare-image/task.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ prepare: |
     echo Running prepare-image
     #shellcheck disable=SC2086
     ARCH="$(dpkg-architecture -qDEB_HOST_ARCH)"
-    su -c "SNAPPY_USE_STAGING_STORE=$SNAPPY_USE_STAGING_STORE snap prepare-image --classic --arch $ARCH --channel $CORE_CHANNEL --snap basic_*.snap  --snap classic-gadget_*.snap $TESTSLIB/assertions/developer1-my-classic-w-gadget.model $ROOT"
+    SNAPPY_USE_STAGING_STORE=$SNAPPY_USE_STAGING_STORE snap prepare-image --classic --arch $ARCH --channel $CORE_CHANNEL --snap basic_*.snap  --snap classic-gadget_*.snap $TESTSLIB/assertions/developer1-my-classic-w-gadget.model $ROOT
```
Comment (bboozzoo) [inline tests/main/classic-prepare-image/task.yaml]: I guess that su wasn't really needed, as it still ran as root.

File: tests/main/classic-prepare-image/task.yaml [hunk 2]
```diff
@@ -52,7 +52,7 @@ restore: |
     systemctl stop snapd.service snapd.socket fakedevicesvc
 
     rm -rf "$SEED_DIR"
-    systemctl start snapd.socket snapd.service
+    systemctl start snapd.socket
```
Comment (bboozzoo) [inline tests/main/classic-prepare-image/task.yaml:55]: I think I'd move this after the fake store is torn down, and "$ROOT" is removed. Also, maybe worth checking if we really need to start snapd in restore. The reset code will stop it anyway.

**Review by bboozzoo — APPROVED** [review]
LGTM

File: tests/main/classic-prepare-image-no-core/task.yaml [hunk 3]
```diff
@@ -35,7 +35,7 @@ prepare: |
     echo Running prepare-image
     #shellcheck disable=SC2086
     ARCH="$(dpkg-architecture -qDEB_HOST_ARCH)"
-    su -c "SNAPPY_USE_STAGING_STORE=$SNAPPY_USE_STAGING_STORE snap prepare-image --classic --arch $ARCH --channel $CORE_CHANNEL --snap basic18_*.snap  --snap classic-gadget-18_*.snap $TESTSLIB/assertions/developer1-my-classic-w-gadget-18.model $ROOT"
+    SNAPPY_USE_STAGING_STORE="$SNAPPY_USE_STAGING_STORE" snap prepare-image --classic --arch "$ARCH" --channel "$CORE_CHANNEL" --snap basic18_*.snap  --snap classic-gadget-18_*.snap "$TESTSLIB"/assertions/developer1-my-classic-w-gadget-18.model "$ROOT"
```
Comment (stolowski) [inline tests/main/classic-prepare-image-no-core/task.yaml:38]: Silly question, why `SNAPPY_USE_STAGING_STORE="$SNAPPY_USE_STAGING_STORE"`? Does it have any effect?

Comment (mvo5) [reply tests/main/classic-prepare-image-no-core/task.yaml:38]: Nice catch, I think this is just a leftover from the "su -c" where this if of courses needed.

File: tests/main/classic-prepare-image/task.yaml (hunk 2 above)
Comment (mvo5) [reply tests/main/classic-prepare-image/task.yaml:55]: Plus we need to apply this to both task.yaml files, right? It seems the other has very similar code and there is no change.

**Comment by mvo5** [comment]
I pushed two small tweak based on the feedback from Maciej and Pawel, let's see if this gets green. Please just revert if I misunderstood or did something stupid.

**Comment by mvo5** [comment]
Hm, either my change broke something or there is more to this. With the latest run I see the following error in the log:
```
...
2021-06-22T09:00:54.1547171Z ##[error]2021-06-22 09:00:53 Error executing google:ubuntu-20.04-64:tests/main/snap-user-service-socket-activation (jun220816-118291) : 
2021-06-22T09:00:54.1556126Z -----
2021-06-22T09:00:54.1557182Z + echo 'When the service snap is installed'
2021-06-22T09:00:54.1558003Z When the service snap is installed
2021-06-22T09:00:54.1559687Z + /home/gopath/src/github.com/snapcore/snapd/tests/lib/tools/snaps-state install-local test-snapd-user-service-sockets
2021-06-22T09:00:54.1560949Z error: cannot perform the following tasks:
2021-06-22T09:00:54.1563200Z - Make snap "test-snapd-user-service-sockets" (unset) available to the system (Post http://0/v1/service-control: dial unix /run/user/0/snapd-session-agent.socket: connect: connection refused)
2021-06-22T09:00:54.1564891Z -----
...
```

File: tests/main/classic-prepare-image/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/main/classic-prepare-image/task.yaml]: yes, that's the idea, thanks for the review

**Comment by sergiocazzolato** [comment]
@mvo5 thanks for pushing those fixes, the test snap-user-service-socket-activation failed because another reason, the tests updated have not been executed before that one. 
 am digging on this new error and I'll create a new PR then.

**Pull request merged**